packages/js/product-editor/src/blocks/product-fields/images/edit.tsx

```diff
@@ -193,6 +194,12 @@ export function Edit( {
 		}
 	}
 
+	const isImageGalleryVisible =
+		propertyValue !== null &&
+		propertyValue !== undefined &&
```

Comment:
You don't need these checks. `Array.isArray()` will return `false` for either `null` or `undefined`.